src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java

```diff
@@ -57,6 +59,38 @@
     return ResponseEntity.ok().body(body);
   }
 
+  @ApiOperation(value = "Update a commons")
+  @PreAuthorize("hasRole('ROLE_ADMIN')")
+  @PutMapping("/update")
+  public ResponseEntity<String> updateCommons(
+    @ApiParam("commons identifier") @RequestParam long id,
+    @ApiParam("request body") @RequestBody CreateCommonsParams params
+  ) throws JsonProcessingException
+  {
+    Optional<Commons> existing = commonsRepository.findById(id);
+
+    Commons updated;
+    HttpStatus status;
+
+    if (existing.isPresent()) {
+      updated = existing.get();
+      status = HttpStatus.NO_CONTENT;
+    } else {
+      updated = new Commons();
+      status = HttpStatus.CREATED;
+    }
```

Comment:
Thank you for following HTTP status code conventions. I don't think we ever covered this explicitly in class, but the attention to detail is nice.